src/librustc_trans/save/mod.rs

```diff
@@ -311,20 +324,23 @@ impl<'l, 'tcx: 'l> SaveContext<'l, 'tcx> {
                     _ => {
                         // Less useful case, impl for a compound type.
                         let span = typ.span;
-                        sub_span = self.span_utils.sub_span_for_type_name(span).unwrap_or(span);
+                        sub_span = self.span_utils.sub_span_for_type_name(span).or(Some(span));
```

Comment:
That's just necessary because filter generated expects subspan to be an Option (in the check just below), but to keep the behaviour correct I still need to swap out the subspan for the span if it's None.